### PR TITLE
Bump Go dependency `emicklei/go-restful`

### PR DESCRIPTION
**Pull request by macedogm**
#### Proposed Changes ####

Bump `emicklei/go-restful` to fix an old CVE.

#### Further Comments ####

Signed-off-by: Guilherme Macedo <guilherme.macedo@suse.com>

### CODE REVIEW

**Comment by macedogm** [comment]
@brandond should I open also against `release-1.22` up to `release-1.25`?

File: go.mod [hunk 1]
```diff
@@ -194,7 +194,7 @@ require (
 	github.com/docker/go-metrics v0.0.1 // indirect
 	github.com/docker/go-units v0.5.0 // indirect
 	github.com/dustin/go-humanize v1.0.0 // indirect
-	github.com/emicklei/go-restful v2.11.2-0.20200112161605-a7c079c43d51+incompatible // indirect
```
Comment (brandond) [inline go.mod:197]: I am curious what we have that's pulling in this dep. upstream Kubernetes is on `go-restful/v3` so there must be something else that's still pulling in the old lib?

Comment (macedogm) [reply go.mod:197]: Yes, it's still being pulled inside RKE2 binary. This is after the update:

```
% go version -m rke2 | grep go-rest
	dep	github.com/emicklei/go-restful	v2.16.0+incompatible	h1:rgqiKNjTnFQA6kkhFe16D8epTksy9HQ1MyrbDXSdYhM=
	dep	github.com/emicklei/go-restful/v3	v3.9.0	h1:XwGDlfxEnQZzuopoqxwSEllNcCOM9DhhFyhFIIGKwxE=
	dep	go.opentelemetry.io/contrib/instrumentation/github.com/emicklei/go-restful/otelrestful	v0.35.0
	=>	go.opentelemetry.io/contrib/instrumentation/github.com/emicklei/go-restful/otelrestful	v0.35.0	h1:KQjX0qQ8H21oBUAvFp4ZLKJMMLIluONvSPDAFIGmX58=
```

It seems to be coming from:

```
% go mod why -m github.com/emicklei/go-restful
# github.com/emicklei/go-restful
github.com/rancher/rke2/pkg/rke2
github.com/k3s-io/k3s/pkg/cli/agent
github.com/k3s-io/k3s/pkg/agent
github.com/k3s-io/k3s/pkg/agent/cridockerd
github.com/Mirantis/cri-dockerd/cmd
github.com/Mirantis/cri-dockerd/streaming
github.com/emicklei/go-restful
```

```
% grep cri-dockerd go.mod 
	github.com/Mirantis/cri-dockerd => github.com/k3s-io/cri-dockerd v0.0.0-20221129193338-7031ab1dbde9
	github.com/Mirantis/cri-dockerd v0.0.0-00010101000000-000000000000 // indirect
```

But both [`k3s-io/cri-dockerd`](https://github.com/k3s-io/cri-dockerd/blob/k3s-master/go.mod#L15) and [`k3s-io/k3s`](https://github.com/k3s-io/k3s/blob/master/go.mod#L230) are pulling `v2.9.5+incompatible` and not `v2.11.2-0.20200112161605-a7c079c43d51+incompatible`. Not easy to find the right place, unless I'm doing something wrong.

Comment (brandond) [reply go.mod:197]: Hmm. We should probably put that behind a build tag on the k3s side, as we're not supporting the --docker flag with RKE2 so there's no reason to pull in cri-dockerd.

Comment (macedogm) [reply go.mod:197]: @brandond can we merge this PR or do we need to wait?

**Comment by brandond** [comment]
Let's hold off for now. If this is only used by cri-dockerd, which RKE2 never enables, then I think the correct fix is to stop building that in, rather than bumping dependencies for things that aren't supposed to be present.

**Comment by caroline-suse-rancher** [comment]
Close due to fix already addressed

**Pull request closed without merging**